src/_pytest/main.py

```diff
@@ -708,11 +713,11 @@ def collect(self) -> Iterator[Union[nodes.Item, nodes.Collector]]:
                         continue
 
                     for x in self._collectfile(path):
-                        key = (type(x), x.fspath)
-                        if key in node_cache2:
-                            yield node_cache2[key]
+                        keya = (type(x), x.fspath)
+                        if keya in node_cache2:
+                            yield node_cache2[keya]
                         else:
-                            node_cache2[key] = x
+                            node_cache2[keya] = x
```

Comment:
I wonder if there's a good way to abstract this cache pattern that's repeated here

~technically this is a little wasteful (it does a double lookup in the "fast" case)